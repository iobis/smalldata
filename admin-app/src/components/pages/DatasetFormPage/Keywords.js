import React from 'react'
import InputMultipleText from '@smalldata/dwca-lib/src/components/form/InputMultipleText'
import PropTypes from 'prop-types'

export default function Keywords({ keywords, onChange }) {
  return (
    <div className="keywords section is-fluid">
      <InputMultipleText
        className="keywords-input"
        name="datasetFormPage.keywords.keywords"
        onChange={onChange}
        values={keywords}/>
    </div>
  )
}

Keywords.propTypes = {
  keywords: PropTypes.arrayOf(PropTypes.string).isRequired,
  onChange: PropTypes.func.isRequired
}
