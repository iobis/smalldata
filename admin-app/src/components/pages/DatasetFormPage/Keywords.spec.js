import Keywords from './Keywords'
import React from 'react'
import { mount } from 'enzyme'

describe('Keywords', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <Keywords
        keywords={[]}
        onChange={jest.fn()}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
