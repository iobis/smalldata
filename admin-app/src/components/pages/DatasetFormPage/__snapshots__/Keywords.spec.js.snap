// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keywords renders correctly 1`] = `
<Keywords
  keywords={Array []}
  onChange={[MockFunction]}
>
  <div
    className="keywords section is-fluid"
  >
    <InputMultipleText
      className="keywords-input"
      name="datasetFormPage.keywords.keywords"
      onChange={[MockFunction]}
      values={Array []}
    >
      <div
        className="column field keywords-input"
      >
        <label
          className="label"
        >
          datasetFormPage.keywords.keywords.label
        </label>
        <div
          className="field has-addons"
        >
          <div
            className="control is-expanded"
          >
            <input
              className="input"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="datasetFormPage.keywords.keywords.placeholder"
              type="text"
              value=""
            />
          </div>
          <div
            className="control"
          >
            <button
              className="add button"
              onClick={[Function]}
            >
              common.add
            </button>
          </div>
        </div>
        <p
          className="help"
        >
          datasetFormPage.keywords.keywords.help
        </p>
        <Tags
          onDelete={[Function]}
          strings={Array []}
        >
          <div
            className="block"
          />
        </Tags>
      </div>
    </InputMultipleText>
  </div>
</Keywords>
`;
